invoice.py

```diff
@@ -52,7 +52,5 @@ def get_price_list(self):
                 prices = Product.get_purchase_price([self.product],
                     self.quantity or 0)
                 self.unit_price = prices[self.product.id]
-        elif self.product:
+        elif not self.unit_price and self.product:
             self.unit_price = self.product.list_price
-        else:
-            self.unit_price = None
```

Comment:
Is it not necessary the last 'else'? Is there no need to set the unit_price, although None?

Reply:
In case the user set a unit_price, and after change the quantity, keep the unit_price.

when change the unit_price in the line?
- party and party.sale_price_list and self.product and invoice_type == 'out'
- party and self.product and invoice_type == 'in' and hasattr(self.product, 'product_suppliers')

In other cases, keep the unit price (not has a product ...) - not set none the unit_price